### PR TITLE
Ncl4672

### PR DESCRIPTION
**Pull request by rnc**
@geoand (WIP to discuss after PTO)

Within this PR I have also refactored the layout slightly and adds a nicer output for the tests.

Its main point was to quieten down the logging so that in PNC its easier to read the alignment log. Things to discuss:
- Should we use Gradle logger ? (https://docs.gradle.org/current/userguide/logging.html)
- Should we output using lifecycle (dependent on above) or info etc.
   - Note that if we need debug logging from the analyser it gets _very_ messy due to the amount pushed out by Gradle - i.e. maybe we should use lifecycle & info. 
- Or potentially create out own wrapper which sends info to llifecycle and debug to info....

### CODE REVIEW

**Comment by geoand** [comment]
Cool, I'll check it out in the next few days!

**Comment by rnc** [comment]
@geoand  Cool :-) I'm leaning towards the first **and** third option right now but we can bounce ideas around when we're both back off holiday. 

I also found https://github.com/wfhartford/gradle-dependency-analyze which seems like a nice equivalent of https://github.com/ImmobilienScout24/illegal-transitive-dependency-check (which I use in PME) and reveals how we are using slf4j-api without declaring it.

**Comment by geoand** [comment]
@rnc looks cool! Let's discuss this more when you are back

**Comment by rnc** [comment]
@geoand  WIP of (1) and (2) together with some colours, better context (and the context ability may be essential for other upcoming PNC requirements).

**Comment by geoand** [comment]
Awesome! Let's discuss once you get back

**Comment by rnc** [comment]
Sure :-) I think its almost ready for a proper review now (!) 

Also , tis hopefully may make it easier to add NCL-5068.

**Comment by geoand** [comment]
@rnc should I start looking at the PR now?

File: build.gradle.kts [hunk 1]
```diff
@@ -74,6 +75,7 @@ subprojects {
         apply(plugin = "java-gradle-plugin")
         apply(plugin = "com.github.johnrengelman.shadow")
         apply(plugin = "com.gradle.plugin-publish")
+        apply(plugin = "com.adarshr.test-logger")
```
Comment (geoand) [inline build.gradle.kts]: The test output looks awesome with this, +100

**Comment by geoand** [comment]
I am seeing a test failure when running locally (`./gradlew clean build`).

```
  org.jboss.gm.analyzer.alignment.AlignmentTaskEmptyVersionTest

    ✘ testProject

      java.lang.reflect.InvocationTargetException
          at org.jboss.gm.analyzer.alignment.AlignmentTaskEmptyVersionTest.testProject(AlignmentTaskEmptyVersionTest.java:96)
      Caused by: org.commonjava.maven.ext.common.ManipulationUncheckedException: For configuration default, unable to resolve all project dependencies: [org.apache.commons:dummy-artifact:]
          at org.jboss.gm.analyzer.alignment.AlignmentTaskEmptyVersionTest.testProject(AlignmentTaskEmptyVersionTest.java:96)

```

When I run from `master` instead of the PR everything is fine. Is it working for you @rnc ?

**Comment by rnc** [comment]
Pushed a fix ( I think!)

**Comment by geoand** [comment]
Tests now pass locally with the latest changes, :+1:

**Pull request merged**